lib/src/main/kotlin/com/swmansion/starknet/data/serializers/GetBlockWithPolymorphicSerializer.kt

```diff
@@ -24,11 +18,22 @@ internal object GetBlockWithTransactionsPolymorphicSerializer : JsonContentPolym
 
 internal object GetBlockWithTransactionHashesPolymorphicSerializer : JsonContentPolymorphicSerializer<GetBlockWithTransactionHashesResponse>(GetBlockWithTransactionHashesResponse::class) {
     override fun selectDeserializer(element: JsonElement): DeserializationStrategy<GetBlockWithTransactionHashesResponse> {
-        val isPendingBlock = listOf("block_hash", "block_number", "new_root").any { it !in element.jsonObject }
-
-        return when (isPendingBlock) {
+        return when (isPendingBlock(element.jsonObject)) {
             true -> PendingBlockWithTransactionHashesResponse.serializer()
             false -> BlockWithTransactionHashesResponse.serializer()
         }
     }
 }
+
+internal object GetBlockWithReceiptsPolymorphicSerializer : JsonContentPolymorphicSerializer<GetBlockWithReceiptsResponse>(GetBlockWithReceiptsResponse::class) {
+    override fun selectDeserializer(element: JsonElement): DeserializationStrategy<GetBlockWithReceiptsResponse> {
+        return when (isPendingBlock(element.jsonObject)) {
+            true -> PendingBlockWithReceiptsResponse.serializer()
+            false -> BlockWithReceiptsResponse.serializer()
+        }
+    }
+}
+
+private fun isPendingBlock(element: JsonObject): Boolean {
+    return listOf("block_hash", "block_number", "new_root").any { it !in element }
```

Comment:
Does it make sense to check that all these keys are missing, not only one of them?

Reply:
This will return `true` if `any` of the keys is missing.

Reply:
Yup, so my question was that maybe we should return `true` iff all the keys are missing.

Reply:
Why though?

If we actually consider the case when there is only one of the values missing in the node's response (which is not something we really should have to account for), and `ignoreUnknownKeys` is set to `true`, the exception will be avoided thanks to the current approach. If not set, deserialization will fail anyway. 

Assuming normal behavior, using `any` just slightly reduces the number of checks we make here 😅

Reply:
I overlooked the fact that deserialization will only work if all keys are missing (without using `ignoreUnknownKeys`). Given this behavior, I am fine with the current approach.